EIPS/eip-4788.md

```diff
@@ -77,76 +79,165 @@ When verifying a block, execution clients **MUST** ensure the root value in the
 
 For a genesis block with no existing parent beacon block root the 32 zero bytes are used as a root placeholder.
 
-### EVM changes
+#### Beacon roots contract
 
-#### Block processing
+The beacon roots contract has two operations: `get` and `set`. The input itself is not used to determine which function to execute, for that the result of `caller` is used. If `caller` is equal to `SYSTEM_ADDRESS` then the operation to perform is `set`. Otherwise, `get`.
```

Comment:
oh, funny. seems okay though

Reply:
This somewhat seems to imply that this contract has an ABI (which it has not, at least not in current assembly). I would really like to do the ABI-like approach as in https://github.com/lightclient/4788asm/pull/5 since this will also be very helpful for solidity users (you can now `BeaconRootContract(address).get(timest)`.

Reply:
I think it is a bad idea to enshrine solidity ABI behavior. It is ubiquitous today, but in the future I expect other language will have different calling conventions. Solidity should directly support this contract like they do for `ecrecover`.

Reply:
sounds like a hack imo